src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.CompilationManager.cs

```diff
@@ -0,0 +1,218 @@
+            {
+                var analyzers = stateSets.Select(s => s.Analyzer).ToImmutableArrayOrEmpty();
+                return CreateAnalyzerDriverAsync(project, analyzers, includeSuppressedDiagnostics, cancellationToken);
+            }
+
+            public async Task<CompilationWithAnalyzers> CreateAnalyzerDriverAsync(
+                Project project, ImmutableArray<DiagnosticAnalyzer> analyzers, bool includeSuppressedDiagnostics, CancellationToken cancellationToken)
+            {
+                if (!project.SupportsCompilation)
+                {
+                    return null;
+                }
+
+                var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+
+                // Create driver that holds onto compilation and associated analyzers
+                return CreateAnalyzerDriver(
+                    project, compilation, analyzers, logAnalyzerExecutionTime: false, reportSuppressedDiagnostics: includeSuppressedDiagnostics);
+            }
+
+            private CompilationWithAnalyzers CreateAnalyzerDriver(
+                Project project,
+                Compilation compilation,
+                ImmutableArray<DiagnosticAnalyzer> analyzers,
+                bool logAnalyzerExecutionTime,
+                bool reportSuppressedDiagnostics)
+            {
+                // PERF: there is no analyzers for this compilation.
+                //       compilationWithAnalyzer will throw if it is created with no analyzers which is perf optimization.
+                if (analyzers.IsEmpty)
+                {
+                    return null;
+                }
+
+                Contract.ThrowIfFalse(project.SupportsCompilation);
+                AssertCompilation(project, compilation);
+
+                var analysisOptions = GetAnalyzerOptions(project, logAnalyzerExecutionTime, reportSuppressedDiagnostics);
+
+                // Create driver that holds onto compilation and associated analyzers
+                return compilation.WithAnalyzers(analyzers, analysisOptions);
+            }
+
+            private CompilationWithAnalyzersOptions GetAnalyzerOptions(
+                Project project,
+                bool logAnalyzerExecutionTime,
+                bool reportSuppressedDiagnostics)
+            {
+                // in IDE, we always set concurrentAnalysis == false otherwise, we can get into thread starvation due to
+                // async being used with syncronous blocking concurrency.
+                return new CompilationWithAnalyzersOptions(
+                    options: new WorkspaceAnalyzerOptions(project.AnalyzerOptions, project.Solution.Workspace),
+                    onAnalyzerException: GetOnAnalyzerException(project.Id),
+                    analyzerExceptionFilter: GetAnalyzerExceptionFilter(project),
+                    concurrentAnalysis: false,
```

Comment:
Is this going to be used for all GetDiagnostics calls. In Update1 we regressed perf for FixAll as we were not computing diagnostics with concurrentAnalysis. Now we explicitly do so: http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/EngineV1/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs,429 for FixAll code path (basically explicit diagnostic requests for project/solution from non-solution crawler path). We should ensure we do so in v2 engine too.

FYI @basoundr @rchande we should probably add a FixAll perf test soon, or we may have to manually verify FixAll perf doesn't regress before Update3.